server/functions/extract.go

```diff
@@ -0,0 +1,205 @@
+		field := val1.(string)
+		if strings.HasPrefix(strings.ToLower(field), "timezone") {
+			return nil, ErrUnitNotSupported.New(field, "time without time zone")
+		}
+		timeVal := val2.(time.Time)
+		return getFieldFromTimeVal(field, timeVal)
+	},
+}
+
+// extract_text_timetz represents the PostgreSQL date/time function, taking {text, time with time zone}
+var extract_text_timetz = framework.Function2{
+	Name:               "extract",
+	Return:             pgtypes.Numeric,
+	Parameters:         [2]pgtypes.DoltgresType{pgtypes.Text, pgtypes.TimeTZ},
+	IsNonDeterministic: true,
+	Strict:             true,
+	Callable: func(ctx *sql.Context, _ [3]pgtypes.DoltgresType, val1, val2 any) (any, error) {
+		field := val1.(string)
+		timetzVal := val2.(time.Time)
+		return getFieldFromTimeVal(field, timetzVal)
+	},
+}
+
+// extract_text_timestamp represents the PostgreSQL date/time function, taking {text, timestamp without time zone}
+var extract_text_timestamp = framework.Function2{
+	Name:               "extract",
+	Return:             pgtypes.Numeric,
+	Parameters:         [2]pgtypes.DoltgresType{pgtypes.Text, pgtypes.Timestamp},
+	IsNonDeterministic: true,
+	Strict:             true,
+	Callable: func(ctx *sql.Context, _ [3]pgtypes.DoltgresType, val1, val2 any) (any, error) {
+		field := val1.(string)
+		if strings.HasPrefix(strings.ToLower(field), "timezone") {
+			return nil, ErrUnitNotSupported.New(field, "timestamp without time zone")
+		}
+		tsVal := val2.(time.Time)
+		return getFieldFromTimeVal(field, tsVal)
+	},
+}
+
+// extract_text_timestamptz represents the PostgreSQL date/time function, taking {text, timestamp with time zone}
+var extract_text_timestamptz = framework.Function2{
+	Name:               "extract",
+	Return:             pgtypes.Numeric,
+	Parameters:         [2]pgtypes.DoltgresType{pgtypes.Text, pgtypes.TimestampTZ},
+	IsNonDeterministic: true,
+	Strict:             true,
+	Callable: func(ctx *sql.Context, _ [3]pgtypes.DoltgresType, val1, val2 any) (any, error) {
+		field := val1.(string)
+		tstzVal := val2.(time.Time)
+		return getFieldFromTimeVal(field, tstzVal)
+	},
+}
+
+//// extract_text_interval represents the PostgreSQL date/time function, taking {text, interval}
+//var extract_text_interval = framework.Function2{
+//	Name:               "extract",
+//	Return:             pgtypes.Numeric,
+//	Parameters:         [2]pgtypes.DoltgresType{pgtypes.Text, pgtypes.Interval},
+//	IsNonDeterministic: true,
+//	Strict:             true,
+//	Callable: func(ctx *sql.Context, _ [3]pgtypes.DoltgresType, val1, val2 any) (any, error) {
+//		field := val1.(string)
+//		if strings.HasPrefix(strings.ToLower(field), "timezone") {
+//			return nil, ErrUnitNotSupported.New(field, "time without time zone")
+//		}
+//		//intervalVal :=
+//		return nil, nil
+//	},
+//}
+
+// getFieldFromTimeVal returns the value for given field extracted from non-interval values.
+func getFieldFromTimeVal(field string, tVal time.Time) (any, error) {
```

Comment:
This should return `decimal.Decimal` to make it very explicit and prevent an accidental type change, since we do not have any safeguards if the wrong type is passed back through the function.